src/customs/useCalculator.ts

```diff
@@ -0,0 +1,65 @@
+import { useState } from 'react'
+
+import { DigitNumber, Operation } from '@/types/calculator'
+
+import { calculator } from '@/utils/calculator'
+
+const useCalculator = () => {
+  const [total, setTotal] = useState(0)
+  const [operator, setOperator] = useState({
+    leftPort: 0,
+    rightPort: 0,
+    expression: '',
+  })
+
+  const handleDigit = (digit: DigitNumber) => {
+    const { expression } = operator
+    const newTotal = Boolean(expression) ? digit : Number(total.toString().concat(digit.toString()))
+    if (newTotal > 999) return alert('세 자리 수 이상 입력할 수 없습니다.')
+    if (Boolean(expression)) {
+      setOperator({ ...operator, rightPort: newTotal })
+    } else {
+      setOperator({ ...operator, leftPort: newTotal })
+    }
+    setTotal(newTotal)
+  }
+
+  const handleOperation = (operation: Operation) => {
+    const { leftPort, rightPort, expression } = operator
+    if (operation === '=') {
+      switch (expression) {
+        case '+':
+          setTotal(calculator['+'](leftPort, rightPort))
+          break
+        case '-':
+          setTotal(calculator['-'](leftPort, rightPort))
+          break
+        case 'X':
+          setTotal(calculator['X'](leftPort, rightPort))
+          break
+        case '/':
+          if (!isFinite(Math.floor(leftPort / rightPort))) alert('오류입니다.')
+          else setTotal(calculator['/'](leftPort, rightPort))
+          break
+        default:
+          return
+      }
```

Comment:
```suggestion
   const result = calculator[expression](leftPort, rightPort)
   if (Number.isFinite(result)) {
     alert("오류입니다")
     return
   }
```

`calculator`함수 시그니쳐를 다 통일시켜놨는데 이를 활용안하니까 너무 아쉽네요 😭 

`expression` 이 '+' | '-' | 'X' | '/' 이외에 값이 들어올 거에 대해서 방어적인 코드를 작성하고 싶으셨던걸까 싶기도 한데,
그건 다른 방법으로도 풀 수 있을거 같아요! 
입력을 받을 때 런타임체크를 잘 하는 방법도 있을 거 같구요, 아니면 result를 만들기 전에 런타임체크를 하는 방법도 있을거 같구요 😄  여러가지 있을거 같아요!